plugins/commands.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const joi = require('joi');
+const boom = require('boom');
+const requestApi = require('request');
+const SCHEMA_COMMAND_NAMESPACE = joi.string().label('Command namespace');
+const SCHEMA_COMMAND_NAME = joi.string().label('Command name');
+const SCHEMA_COMMAND_VERSION = joi.string().label('Command version');
+const DEFAULT_TTL = 24 * 60 * 60 * 1000; // 1 day
+const DEFAULT_BYTES = 1024 * 1024 * 1024; // 1GB
+
+/**
+ * Determine if a pipeline can publish the command
+ * @method canPublishCommand
+ * @param  {String}  namespace  Command namespace
+ * @param  {String}  name       Command name
+ * @param  {Integer} pipelineId pipelineId
+ * @param  {String}  baseurl    API URL
+ * @return {Promise}
+ */
+function canPublishCommand(namespace, name, pipelineId, baseurl) {
+    return new Promise((resolve, reject) => {
+        requestApi({
+            uri: `${baseurl}/v4/commands`,
+            method: 'GET',
+            json: true
+        },
+        (err, response) => {
+            if (err) {
+                return reject(err);
+            }
+
+            return resolve(response);
+        });
+    }).then((response) => {
+        const commands = response.body.filter(element =>
+            element.namespace === namespace && element.name === name
+        );
+
+        // If commands.length === 0, it is first time to publish namespace/name command.
+        if (commands.length === 0 || commands[0].pipelineId === pipelineId) {
```

Comment:
In the case of `commands.length === 0`, you still allow pushing binaries to S3 but there is no record in the API. Then this binary is not discoverable in the UI/API.